src/cursor.rs

```diff
@@ -387,10 +389,17 @@ impl Cursor {
         let read_preference =
             read_pref.unwrap_or_else(|| ReadPreference::new(ReadMode::Primary, None));
 
+        // Check if actual batch size fits into an `i32`.
+        if size_of::<i32>() <= size_of::<usize>() && buf.len() > i32::MAX as usize {
```

Comment:
I'm struggling to think of when the first clause of this boolean expression won't be true; does Rust actually support any platforms that aren't 32-bit or 64-bit? I suppose this bit of defensive programming won't hurt, but I'd guess that there would be a lot of other issues if someone tried to run this on a 16-bit platform.

Reply:
The [official docs](https://forge.rust-lang.org/platform-support.html) say that Rust has Tier 3 support for 16-bit MSP430 µCs. I agree that those are not the sort of devices people will run MongoDB on (although never say never, look what they've done to JavaScript 😛) – still, I feel that expressing the *actual* constraints (the number fits based on whether or not `usize` is narrower than 32 bits) is clearer than making a seemingly arbitrary "32-vs-64" decision. I think of Rust as a highly portable language, and I just didn't want this to be one more issue should anyone ever run the code on a 16-bit platform.

Reply:
Makes sense. As for people not running MongoDB on tiny microcontrollers, it's worth noting that the driver will almost never be run on the same machine as the server, and Rust is one of the few languages that's well-suited for running a database client on a microcontroller (although this driver isn't suitable for that at this point IMO).